### PR TITLE
Highlight top buttons

### LINKED ISSUE #364: Highlight menu items that are active
I honestly have no idea how this is supposed to work, but it would be a really awesome feature. Just now I was fanatically clicking all of them to find the one I closed

### PR DESCRIPTION
**Pull request by grilledham**
When the gui is open highlight the top button
image

Closes #364

The sprite does grow slightly in size when highlighted. Not sure why I tried setting the padding the same before and after and didn't change it. But maybe it's not a problem

### CODE REVIEW

File: features/gui/info.lua [hunk 1]
```diff
@@ -652,13 +661,20 @@ end
...
+    local center = gui.center
     local main_frame = center[main_frame_name]
+    local main_button = gui.top[main_button_name]
 
     if main_frame then
         upload_changelog(event)
         Gui.destroy(main_frame)
+        main_button.style = 'icon_button'
     else
+        main_button.style = 'selected_slot_button'
```
Comment (linaori) [inline features/gui/info.lua:677]: it's easy to forget adding this when you add a button to the top, it's also error prone as it's pretty much copy-paste. Would it be possible to create a small API where we can talk to, to use this "toggle" feature as it's more than just rendering a window or not right now? 

Perhaps @Cooldude2606 has an idea for this as he made something really nice for something like this already

Comment (grilledham) [reply features/gui/info.lua:677]: I agree this is not ideal.
The problem comes down to how does the button know if it should be in the toggled state or not.
I don't have a solution to that. But I'm open to discuss ideas.

Comment (Cooldude2606) [reply features/gui/info.lua:677]: Toggled states are hard, I have one instance in my code when I use them, the way in which I check if it should be toggled is by using my store module; but the general idea I can still explain.
If you have a table that stores the current "state" of the toggle then you can place the button inside a "boarderless flow" (flow with no padding) and set either its name or caption (both work) to be the key to look up in the "toggle table" that you store and get the state for that button. See my player list action button for my one example.
A note on it chaning size: it isnt chaning size it is just that the slot buttons are made to go next to each other where as the other button styles have a built-in effect which makes them appear to pop out slightly. To make them appear the same size try putting it in a flow, adding some padding but then making the button smaller.

Comment (grilledham) [reply features/gui/info.lua:677]: I refactored the modules that had multiple ways in which the main frame could be closed to have a single `close_main_frame` function. Without a non minor change to way we write the gui code I don't think we can do any better.

Regarding the icon size change:
I'm not sure if I misunderstood you cooldude, but the button itself stays the same size, it's the sprite inside the button that changes size. 

I can live with the sprite changing size, however if anyone wants to look into it they can.

Comment (Cooldude2606) [reply features/gui/info.lua:677]: Sorry, I thought you meant that the size the the button looked was different without the size being different. I have had this issue before which is why I thought you meant this.

**Pull request merged**